src/main/java/racingcar/model/RacingCar.java

```diff
@@ -4,56 +4,51 @@
 import racingcar.model.movestrategy.RandomVarMoveStrategy;
 
 public class RacingCar {
-    private static final String ERR_NAME_LENGTH = "Names longer than 5 characters are not allowed.";
-    private static final int NAME_MAX_LENGTH = 5;
-
-    private MoveStrategy moveStrategy;
+    private final CarName name;
+    private final MoveStrategy moveStrategy;
     private Position position;
-    private String name;
 
     public RacingCar(String name) {
-        // set MoveStrategy
-        this.moveStrategy = new RandomVarMoveStrategy();
-
-        // validation
-        validateCarName(name);
-
-        // setting
-        this.name = name;
-        this.position = new Position();
+        this(new RandomVarMoveStrategy(), name);
     }
 
     public RacingCar(MoveStrategy moveStrategy, String name) {
         // set MoveStrategy
         this.moveStrategy = moveStrategy;
 
-        // validation
-        validateCarName(name);
-
         // setting
-        this.name = name;
+        this.name = new CarName(name);
         this.position = new Position();
     }
 
-    private void validateCarName(String name) {
-        if (name.length() > NAME_MAX_LENGTH) {
-            throw new IllegalArgumentException(ERR_NAME_LENGTH);
+
+    public void moveOrStop() {
+        if (moveStrategy.canMove()) {
+            this.move();
         }
     }
 
-
-    public void moveOrStop(int value) {
-        if (moveStrategy.canMove(value)) {
-            this.position.step();
-        }
+    private void move() {
+        this.position = new Position(this.position.step());
```

Comment:
position 객체에게 메시지를 보내도 좋을 것 같아요
```suggestion
        this.position = this.position.increase();
```